packages/core/src/browser/preferences/preference-service.ts

```diff
@@ -27,11 +32,12 @@ export interface PreferenceService extends Disposable {
     get<T>(preferenceName: string): T | undefined;
     get<T>(preferenceName: string, defaultValue: T): T;
     get<T>(preferenceName: string, defaultValue?: T): T | undefined;
+    set(preferenceName: string, value: string, scope?: PreferenceScope): void;
     onPreferenceChanged: Event<PreferenceChange>;
 }
 
-export const PreferenceProviders = Symbol('PreferenceProvidersFactory');
-export type PreferenceProviders = () => PreferenceProvider[];
+export const PreferenceProviderFactory = Symbol('PreferenceProviderFactory');
+export type PreferenceProviderFactory = (scope: PreferenceScope) => PreferenceProvider;
```

Comment:
please rename it to `PreferenceProviders`. 
inversify's `toFactory` is a bit misleading, as not every synchronous function is a factory.

Reply:
`PreferenceProviders` sounds like a collection. How about rename existing `PreferenceProvider` to just `Preferences` and then rename this to `PreferenceProvider`?

Reply:
How about completely get rid of it and inject user and workspace providers directly? It should simplify binding configuration and implementation.

Reply:
@AlexTugarev 
```
@inject(PreferenceProvider) @named(PreferenceScope.Workspace)
protected readonly workspacePreferences: PreferenceProvider;
```
If this present in `PreferenceServiceImpl` it causes an error when launching Theia:
![20171013_183624](https://user-images.githubusercontent.com/7668752/38916084-b381371e-42ee-11e8-92a7-7bc686d403c1.jpg)

Reply:
@vinokurig, could you share that change? a different branch would be fine.

Reply:
@AlexTugarev https://github.com/vinokurig/theia/tree/StorePreferences_1